test/mocking_classes.py

```diff
@@ -1700,6 +1748,7 @@ def _step(
         self.count += one_hot_action.to(torch.int)
         td["observation"] += expand_right(self.count, td["observation"].shape)
         done["done"] = self.count > self.max_steps
+        done["terminated"] = self.count > self.max_steps
```

Comment:
because we want done to be able to quickly know if we should reset. It avoids always going through all the possible keys and carries the summary of the end-of-traj in one tensor. 
So now all envs have at least one done and one terminated, possibly a truncated.
It's what we discussed with @skandermoalla IIRC.

Reply:
but the done would be added automatically by EnvBase no?

Reply:
Not sure if my input is needed here, but here's how I understand this:
>  So now all envs have at least one done and one terminated, possibly a truncated.

Yes, this is what is expected.

> but the done would be added automatically by EnvBase no?
i am talking about this mocking class, it could just output terminated and done would be built

Yes, the done key can be inferred from the terminated (possibly truncated keys). I haven't looked at the implementation yet to know where this is done, but if the EnvBase takes care of this, then this mock can avoid writing the done key indeed to test the automatic done inference feature.

Reply:
I think it's best if we show both whenever we can. We can copy the done if you prefer but not showing that it's populated isn't great from a UX perspective IMO
The auto patch is supposedly just there to ensure that everything runs fine even if someone just writes a terminated/done key

Reply:
just did, what did I miss?